examples/cc3200/xively_demo/main.c

```diff
@@ -0,0 +1,1127 @@
+ */
+void onLedTopic( xi_context_handle_t in_context_handle,
+                 xi_sub_call_type_t call_type,
+                 const xi_sub_call_params_t* const params,
+                 xi_state_t state,
+                 void* user_data )
+{
+    switch ( call_type )
+    {
+        case XI_SUB_CALL_SUBACK:
+            if ( params->suback.suback_status == XI_MQTT_SUBACK_FAILED )
+            {
+                UART_PRINT( "topic:%s. Subscription failed.\n", params->suback.topic );
+            }
+            else
+            {
+                UART_PRINT( "topic:%s. Subscription granted %d.\n", params->suback.topic,
+                            ( int )params->suback.suback_status );
+            }
+            return;
+        case XI_SUB_CALL_MESSAGE:
+        {
+            if ( params->message.temporary_payload_data_length == 1 )
+            {
+                ledNames ledName = *( ( ledNames* )user_data );
+
+                switch ( params->message.temporary_payload_data[0] )
+                {
+                    case 48:
+                        UART_PRINT( "topic:%s off %d\r\n", params->suback.topic,
+                                    ( int )ledName );
+                        GPIO_IF_LedOff( ledName );
+                        break;
+                    case 49:
+                        UART_PRINT( "topic:%s on %d\r\n", params->suback.topic,
+                                    ( int )ledName );
+                        GPIO_IF_LedOn( ledName );
+                        break;
+                    default:
+                        UART_PRINT( "unexpected value on topic %s \r\n",
+                                    params->message.topic );
+                        break;
+                }
+            }
+            else
+            {
+                UART_PRINT( "unexpected data length on topic %s \r\n",
+                            params->message.topic );
+            }
+        }
+            return;
+        default:
+            return;
+    }
+}
+
+/**
+ * @brief Implementation of recurring time task that sends updates on temperature sensor
+ * on each invocation.
+ *
+ * @note for more details about user time tasks please refer to the xively.h -
+ * xi_schedule_timed_task function.
+ *
+ * @param context_handle - xively library context required for calling API functions
+ * @param timed_task_handle - handle to *this* timed task, can be used to re-schedule or
+ * cancel the task
+ * @param user_data - data assosicated with this timed task
+ */
+void send_temprature( const xi_context_handle_t context_handle,
+                      const xi_timed_task_handle_t timed_task_handle,
+                      void* user_data )
+{
+    // Prepare configuration for temperature sensor for I2C bus
+    PinTypeI2C( PIN_01, PIN_MODE_1 );
+    PinTypeI2C( PIN_02, PIN_MODE_1 );
+
+    int lRetVal = -1;
+
+    // I2C Init
+    lRetVal = I2C_IF_Open( I2C_MASTER_MODE_FST );
+    if ( lRetVal < 0 )
+    {
+        ERR_PRINT( lRetVal );

```

Comment:
you should set the pin mode back to the button configuration on errors.  you could put the PinType calls into a function and just call the function on each error state.